src/rust/scheduler/scheduler.rs

```diff
@@ -402,39 +421,82 @@ mod tests {
         Ok(())
     }
 
-    /// Tests if consecutive tasks are not assigned the same task id.
     #[test]
-    fn test_scheduler_task_ids() -> Result<()> {
```

Comment:
I am having a hard time to understand how this is tested now. Have we dropped this test?